client/src/modules/cohort/index.jsx

```diff
@@ -66,33 +71,71 @@ class Cohort extends PureComponent {
         params: { id: cohortId }
       }
     } = this.props;
+    const { isUnavailable } = this.state;
 
     this.fetchData();
-    joinRoom(Routes.COHORT, cohortId, userId);
+
+    const roomConfig = {
+      subscribeMetaData: !isUnavailable,
+      resourceId: cohortId,
+      roomPrefix: Routes.COHORT,
+      userId
+    };
+
+    joinRoom(roomConfig);
   }
 
   componentDidUpdate(previousProps) {
     const {
       cohortDetails: cohort,
       match: {
-        params: { id }
-      }
+        params: { id: cohortId }
+      },
+      members
     } = this.props;
     const {
       cohortDetails: previousCohort,
       currentUser: { id: userId },
       match: {
-        params: { id: previousId }
-      }
+        params: { id: previousCohortId }
+      },
+      members: previousMembers
     } = previousProps;
-
-    if (id !== previousId) {
-      leaveRoom(Routes.COHORT, previousId, userId);
+    const { isUnavailable } = this.state;
+    const hasCohortChanged = cohortId !== previousCohortId;
+    const cohortUpdated = previousCohort && cohort;
+    const roomConfig = {
+      subscribeMetaData: !isUnavailable,
+      resourceId: previousCohortId,
+      roomPrefix: Routes.COHORT,
+      userId
+    };
+
+    if (hasCohortChanged) {
+      leaveRoom(roomConfig);
+      joinRoom({ ...roomConfig, resourceId: cohortId });
       this.fetchData();
     }
 
-    if (previousCohort && cohort && previousCohort.name !== cohort.name) {
-      this.handleBreadcrumbs();
+    if (cohortUpdated) {
+      const updateBreadcrumbs = previousCohort.name !== cohort.name;
+      const hasCohortBeenArchived =
+        !previousCohort.isArchived && cohort.isArchived;
+      const hasCohortBeenRestored =
+        previousCohort.isArchived && !cohort.isArchived;
+      const hasUserBeenRemoved = previousMembers[userId] && !members[userId];
+      const hasCohortBeenDeleted =
+        !previousCohort.isDeleted && cohort.isDeleted;
+
+      if (updateBreadcrumbs) {
+        this.handleBreadcrumbs();
+      }
+
+      if (hasCohortBeenArchived || hasUserBeenRemoved || hasCohortBeenDeleted) {
+        this.handleMakeCohortUnavailable();
+      } else if (hasCohortBeenRestored) {
```

Comment:
simplify to `else`

Reply:
It can not be simplified to else because when cohort is set as unavailable `didUpdate` is called (state was changed) and makes this cohort available again